service/src/event/service.rs

```diff
@@ -32,4 +69,222 @@ impl CeramicEventService {
+            let deliverable = insert_now
+                .iter()
+                .filter_map(|e| {
+                    if e.body.deliverable {
+                        Some(e.body.cid)
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            self.verify_history_inline(
+                insert_after_history_check,
+                &mut insert_now,
+                &mut notify_task_new,
+                &deliverable,
+            )
+            .await?;
+        }
+        let res = CeramicOneEvent::insert_many(&self.pool, &insert_now[..]).await?;
+
+        for ev in background_task_deliverable {
+            trace!(cid=%ev.0, prev=%ev.1.prev, init=%ev.1.init_cid, "sending to delivery task");
+            if let Err(e) = self
+                .delivery_task
+                .tx
+                .try_send(DeliverableEvent::new(ev.0, ev.1, None))
+            {
+                match e {
+                    tokio::sync::mpsc::error::TrySendError::Full(e) => {
+                        // we should only be doing this during recon, in which case we can rediscover events.
+                        // the delivery task will start picking up these events once it's drained since they are stored in the db
+                        warn!(cid=%e.cid, meta=?e.meta, limit=%PENDING_DELIVERABLE_EVENTS, "Delivery task full. Dropping event and will not be able to mark deliverable until queue drains");
+                    }
+                    tokio::sync::mpsc::error::TrySendError::Closed(_) => {
+                        warn!("Delivery task closed. shutting down");
+                        return Err(Error::new_fatal(anyhow::anyhow!("Delivery task closed")));
+                    }
+                }
+            }
+        }
+        for new in notify_task_new {
+            if let Err(e) = self.delivery_task.tx_new.try_send(new) {
+                match e {
+                    tokio::sync::mpsc::error::TrySendError::Full(ev) => {
+                        // we should only be doing this during recon, in which case we can rediscover events.
+                        // the delivery task will start picking up these events once it's drained since they are stored in the db
+                        warn!(attempt=?ev, limit=%PENDING_DELIVERABLE_EVENTS, "Notify new task full");
+                    }
+                    tokio::sync::mpsc::error::TrySendError::Closed(_) => {
+                        warn!("Delivery task closed. shutting down");
+                        return Err(Error::new_fatal(anyhow::anyhow!("Delivery task closed")));
+                    }
+                }
+            }
+        }
+
+        Ok(res)
+    }
+
+    async fn verify_history_inline(
+        &self,
+        insert_after_history_check: Vec<(DeliverableMetadata, EventInsertable)>,
+        insert_now: &mut Vec<EventInsertable>,
+        notify_new: &mut Vec<DeliveredEvent>,
```

Comment:
this is annoying to duplicate. we need to know about the init/stream ID in one of these and not the other. I think it can be simplified by adding a bit more ceramic  info to things after parsing blocks (it was very much just recon K,V before)